### PR TITLE
feat(local): add Thermal reading

### PR DESCRIPTION
**Pull request by killian-scalian**
- Thermal reading
-  Unitary test
- Co2Cost, fuelCost, data, modulation and series timeseries handled

### CODE REVIEW

File: tests/antares/services/local_services/test_area.py [hunk 1]
```diff
@@ -1306,3 +1307,76 @@ def test_read_misc_gen_local(self, local_study_w_areas):
             _write_file(file_path, expected_time_serie)
             matrix = area.get_misc_gen_matrix()
             pd.testing.assert_frame_equal(matrix, expected_time_serie)
+
+
+class TestReadThermal:
+    def create_file_with_empty_dataframe(self, file_path: str, df: pd.DataFrame):
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:1313]: You don't use this method, you can remove it

File: src/antares/service/local_services/thermal_local.py [hunk 2]
```diff
@@ -31,7 +39,74 @@ def update_thermal_properties(
...
+                    op2=thermal_dict[thermal_cluster]["op2"],
+                    op3=thermal_dict[thermal_cluster]["op3"],
+                    op4=thermal_dict[thermal_cluster]["op4"],
+                    op5=thermal_dict[thermal_cluster]["op5"],
+                    cost_generation=thermal_dict[thermal_cluster]["costgeneration"],
+                    efficiency=thermal_dict[thermal_cluster]["efficiency"],
+                    variable_o_m_cost=thermal_dict[thermal_cluster]["variableomcost"],
+                )
+
+            thermal_clusters.append(
```
Comment (MartinBelthle) [inline src/antares/service/local_services/thermal_local.py:104]: Put this inside the loop, it's not indented correctly

File: src/antares/service/local_services/thermal_local.py [hunk 3]
```diff
@@ -31,7 +39,74 @@ def update_thermal_properties(
...
+            cluster_id=thermal_cluster.properties.group.value,
+        )
 
     def read_thermal_clusters(self, area_id: str) -> List[ThermalCluster]:
-        raise NotImplementedError
+        thermal_dict = IniFile(self.config.study_path, IniFileTypes.THERMAL_LIST_INI, area_name=area_id).ini_dict
+        thermal_clusters = []
+        if thermal_dict:
+            for thermal_cluster in thermal_dict:
+                thermal_properties = ThermalClusterPropertiesLocal(
```
Comment (MartinBelthle) [inline src/antares/service/local_services/thermal_local.py:65]: This is really really ugly. We should introduce aliases inside the `ThermalClusterProperitesLocal` so inside the `DefaultThermalProperties` class to use pydantic to validate this. And this goes with adding the aliases camel case inside the `API` class.
But this is too complicated for this PR so you can just add a `TODO` saying that we should refactor this

File: src/antares/service/local_services/thermal_local.py [hunk 4]
```diff
@@ -31,7 +39,74 @@ def update_thermal_properties(
...
+            time_serie_type = TimeSeriesFileType.THERMAL_SERIES
+        elif ts_name.value == "modulation":
+            time_serie_type = TimeSeriesFileType.THERMAL_MODULATION
+        elif ts_name.value == "data":
+            time_serie_type = TimeSeriesFileType.THERMAL_DATA
+        elif ts_name.value == "CO2Cost":
+            time_serie_type = TimeSeriesFileType.THERMAL_CO2
+        elif ts_name.value == "fuelCost":
+            time_serie_type = TimeSeriesFileType.THERMAL_FUEL
+
```
Comment (MartinBelthle) [inline src/antares/service/local_services/thermal_local.py:52]: You should either put a else: raise NotImplementedError here or a else with the fuelCost as here we can reach a null pointer as `time_serie_type` could be undefined

**Review by MartinBelthle — CHANGES_REQUESTED** [review]

File: src/antares/service/local_services/thermal_local.py (hunk 3 above)
Comment (killian-scalian) [reply src/antares/service/local_services/thermal_local.py:66]: Ok, for the last commit

File: src/antares/service/local_services/thermal_local.py (hunk 2 above)
Comment (killian-scalian) [reply src/antares/service/local_services/thermal_local.py]: Ok, for the last commit

File: tests/antares/services/local_services/test_area.py (hunk 1 above)
Comment (killian-scalian) [reply tests/antares/services/local_services/test_area.py]: Ok, for the last commit

**Pull request merged**